### DISCUSSION TITLE
glTF Working Group Meeting 2021-11-11

### DISCUSSION THREAD

**Discussion by humbletim** [omigroup/gltf-extensions#33] [General]
# Agenda

- Update on Implementation Status
  - Spec
    - Need to add markdown explainer
    - Publish blog post advertising the OMI_audio_emitter ecosystem
  - omi-gltf-samples-repo
    - should we set this up?
    - refs:
      - https://github.com/KhronosGroup/glTF-Sample-Models 
      - jin's comments re: samples repo (discord message: https://discord.com/channels/770382203782692945/835369558541467668/907996893039177739)
  - three-omi https://github.com/omigroup/three-omi
  - Third Room https://github.com/thirdroom/thirdroom/pull/17
  - Unity https://github.com/omigroup/glTFast-demo
    - VRChat
    - Altspace
  - Godot
- Next extensions
  - When is it appropriate to start working on these?
  - What resources would make this process easier?
    - File issues for various resources needed
  - What proposals would people like to see most?
- Introduction Kit for using OMI Audio Emitter 
  - ultra simple first impression self-contained .gltf example? (for those wanting a quick glimpse at what integration would be up against)
  - links to resources for integrating
  - links to sample files to test integration with  
  - ...

## Parking Lot
  - Neos
  - Hubs (https://github.com/mozilla/hubs/pull/4741  
  - Stereo Kit

**Comment by fire** [comment]
1. Mike | Icosa worked on the json parser on Unity.

**Reply by fire** [reply]
I have done some work on the MaterialX extension.

1. draft https://github.com/fire/glTF/blob/materialx/extensions/2.0/Vendor/OMI_texture_mtlx/README.md
2. https://github.com/fire/MaterialX/tree/gltf

**Reply by mikeskydev** [reply]
I have been comparing different JSON loaders inside Unity. @atteneder mentioned performance and size concerns for Newtonsoft.Json, which I have confirmed in testing. 

I downloaded the NuGet package for System.Json, which is a platform extension package https://www.nuget.org/packages/System.Json. It has a similar parse speed based on my testing, but System.Json is a much smaller package compared to Newtonsoft.Json. System.Json doesn't need a struct passed to it, and you can query the parsed object for any arbitrary key: `data["node"]` or check if a node exists: `ContainsKey("node")`

UniVRM's parser, called UniJson, came in slightly faster than both. https://github.com/vrm-c/UniVRM/tree/master/Assets/UniGLTF/Runtime/UniJSON

See below for speed comparison on a fairly complex glTF.
![image](https://user-images.githubusercontent.com/9121007/141383428-6a336771-c0e7-4309-bc6d-3030f845db0f.png)

I also want to investigate new .NET 5 (.NET6?) replacement for System.Json, but I'm not sure how compatible the latter is with Unity due to it's .NET version.

**Comment by lyuma** [comment]
Godot status:
Godot's proposed GLTF extension api ( https://github.com/godotengine/godot-proposals/issues/3305 ) has not been finalized / approved yet.
We could move forward with an importer more like the VRM importer. Question: does Godot 3.4 / 4.0 allow overriding the built-in GLTF import?

**Comment by mrmetaverse** [comment]
Suggestions per our conversation: Maybe make an empty node with a audio emitter attached to it with a default sound. Maybe also make an example object, like a sword, and give it a metallic sound. Or a cat glb with a built in purr sound.
